core/postgis_finder.py

```diff
@@ -0,0 +1,182 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+#---------------------------------------------------------------------
+
+from PyQt4.QtCore import QObject, QSettings
+
+from qgis.core import QgsGeometry, QgsCredentials
+from qgis.gui import QgsMessageBar
+
+from my_settings import MySettings
+
+import psycopg2
+import binascii
+
+from postgis_search import PostgisSearch
+from .abstract_finder import AbstractFinder
+
+try:
+    from processing.tools.postgis import uri_from_name, DbError
+except ImportError:
+    # Older processing versions don't have this helper
+    # PostGIS finder won't work.
+    pass  # Don't prevent quickfinder plugin from loading.
+
+
+class PostgisFinder(AbstractFinder):
+
+    name = 'postgis'
+
+    @property
+    def searches(self): return self._searches
+
+    def __init__(self, parent):
+        super(PostgisFinder, self).__init__(parent)
+        self._searches = self.readSearches()
+
+    def start(self, to_find, bbox=None):
+        super(PostgisFinder, self).start(to_find, bbox)
+        dbConnectionName = self.settings.value('postgisConnection')
+        self.cur = None
+        if dbConnectionName:
+            try:
+                connectionUri = uri_from_name(dbConnectionName)
+                self.cur = self.connectToUri(connectionUri)
+            except DbError as err:
+                self.message.emit(unicode(err), QgsMessageBar.WARNING)
+            if self.cur:
+                self.find(to_find)
+        self._finish()
+
+    def connectToUri(self, uri):
+        """Create a connection from a uri and return a cursor of it."""
+        conninfo = uri.connectionInfo()
+        conn = None
+        cur = None
+        ok = False
+        while not conn:
+            try:
+                conn = psycopg2.connect(uri.connectionInfo())
+                cur = conn.cursor()
+            except psycopg2.OperationalError as e:
+                (ok, user, passwd) = QgsCredentials.instance().get(
+                    conninfo, uri.username(), uri.password())
+                if not ok:
+                    break
+
+        if not conn:
+            self.message.emit(
+                "Could not connect to PostgreSQL database",
+                QgsMessageBar.WARNING)
+
+        if ok:
+            QgsCredentials.instance().put(conninfo, user, passwd)
+
+        return cur
+
+    def find(self, to_find):
+        catLimit = self.settings.value("categoryLimit")
+        totalLimit = self.settings.value("totalLimit")
+        hasProjectSearches = len(self.settings.value("postgisSearches"))
+        catFound = {}
```

Comment:
not sure, but you don't handle categories here?
it would be great to be able to specifiy either a column as the category or the layer name itself.